tests/nn/test_module.py

```diff
@@ -217,6 +217,39 @@ def test_constraints(shape, constraint_):
     assert not hasattr(module, 'x_unconstrained')
 
 
+def test_clear():
+
+    class Model(PyroModule):
+        def __init__(self):
+            super().__init__()
+            self.x = nn.Parameter(torch.tensor(0.))
+            self.m = torch.nn.Linear(2, 3)
+            self.m.weight.data.fill_(1.)
+            self.m.bias.data.fill_(2.)
+            self.p = PyroModule()
+            self.p.x = nn.Parameter(torch.tensor(3.))
+
+        def forward(self):
+            return [x.clone() for x in [self.x, self.m.weight, self.m.bias, self.p.x]]
+
+    m = Model()
+    state0 = m()
+
+    # mutate
+    for x in pyro.get_param_store().values():
+        x.unconstrained().data += torch.randn(())
+    state1 = m()
+    for x, y in zip(state0, state1):
+        assert not (x == y).all()
+
+    clear(m)
+    del m
```

Comment:
This will leave `x` in the param store because `clear` only remove root module's Pyro params and its submodules, while `x` is a `nn.Parameter` here. Is it expected? If so, it would be clearer to add `assert len(pyro.get_param_store()) == 1` here.
On the other hand, because all parameters of `Model` are nn.Parameter, `clear` and `del` here are not necessary. I think that defining `self.p.x = pyro.nn.PyroParam(torch.tensor(3.))` would be a better test.

Reply:
Good catch! I've fixed and strengthened tests.